### ISSUE TITLE
React Camelcase

### ISSUE THREAD

**Issue by chrismcleod** [launchdarkly/ld-find-code-refs#86]
The react sdk camelcases hyphenated flag keys and this utility seems to only scan for the hyphenated key which it will not find.

i.e. flag with key `my-key` must be used in react as `const {myKey} = useFlags()`

**Comment by parkerself22** [comment]
Was about to open this issue as well, working around it using comments for now but that is definitely not the best solution.

**Comment by chrismcleod** [comment]
I would like to be able to do

```
const flag = useFlags('my-key')
```

though this can also work if react wasnt mapping these to camelcase, but is less ideal
```
const {['my-key']: flag} = useFlags()
```

**Comment by chrismcleod** [comment]
@parkerself22 i made this custom hook to work around it:

```
export const useFlag = (variation) => {
  const ldClient = useLDClient()
  return ldClient.variation(variation)
}
```

**Comment by parkerself22** [comment]
Thats what I was about to say, we did something similar. Would be nice for this lib to support it by default but its not a terrible workaround

**Comment by vlymar** [comment]
Hey @chrismcleod & @parkerself22, thanks for reporting this.

We're going to try to address this issue in the next few weeks. Unfortunately for now you'll need to use one of the workarounds you've mentioned to get code references to match.

**Comment by parkerself22** [comment]
@chrismcleod (and anyone else who runs into this) had a bit of a facepalm moment when I realized you can also just manually overwrite the default slug generation when creating the tag:

![image](https://user-images.githubusercontent.com/24485514/62415600-8176aa80-b5fa-11e9-9cc3-100a9549240d.png)

A setting to choose the slug generation strategy so you don't have to do this manually every time would be an awesome feature. @vlymar not sure who the person to send that feature request to is but seems like plenty of people would take advantage of it

**Comment by vlymar** [comment]
Good point @parkerself22. Out of curiosity, besides getting ld-find-code-refs to work at the moment, what other use cases would a "use camelcase for flag keys" setting have for you?

**Comment by parkerself22** [comment]
@vlymar Besides ld-find-code-refs it also eliminates the mental overhead of having to figure out what the camelcase version of the key will be when using the JS sdk's. That sounds trivial but it is an easy way to make a dumb mistake when accessing a flag in code. 

For now I'm asking everyone using flags on our team to manually camelcase them, if it is the default that eliminates having to enforce the policy as well.

**Comment by bwoskow-ld** [comment]
Hi @chrismcleod and @parkerself22,

We released [version 2.13.0](https://github.com/launchdarkly/js-client-sdk/releases/tag/2.13.0) of the React SDK to address this issue. You can now provide a configuration option to disable the automatic camel-casing functionality. When disabled, keys will be accessed through the React SDK in their native form (as known on app.launchdarkly.com) -- for example, as `my-key` instead of `myKey`. The option will be enabled by default for backwards compatibility.

You can read more about flag keys in [our documentation](https://docs.launchdarkly.com/docs/react-sdk-reference#section-flag-keys).

With this change in place, we don't have any plans to adapt the code references tool to search for camel-cased key references.

Thanks for reporting this issue!

**Comment by parkerself22** [comment]
@bwoskow-ld Awesome, thanks for rolling out a fix so quickly. Think you found the best possible solution, the expanded explanation in the documentation really explains the problem well.
